tests/utilities/test_reader.py

```diff
@@ -941,3 +941,20 @@ def test_gif_reader(gif_file):
         data_name = f'frame{i}'
         new_grid.point_data.set_array(data, data_name)
         assert np.allclose(grid[data_name], new_grid[data_name])
+
+
+@pytest.mark.skipif(pyvista.vtk_version_info < (9, 1), reason="Requires VTK v9.1.0 or newer")
+def test_xdmf_reader():
+    filename = examples.download_dolfinx_mesh(load=False)
+    reader = pyvista.get_reader(filename)
+    assert isinstance(reader, pyvista.XdmfReader)
+    assert reader.path == filename
+
+    mesh = reader.read()
+    assert all([mesh.n_points, mesh.n_cells])
+
+
+@pytest.mark.skipif(pyvista.vtk_version_info < (9, 1), reason="Requires VTK v9.1.0 or newer")
+def test_xdmf_timepoints():
+    # TODO Add a time points test using meshio/out.xdmf
+    assert True
```

Comment:
Sorry. I cannot work on this now. We need to add a test of time points. Any good time-line xdmf files would be welcome.

Reply:
@tkoyama010 

The files i provided here were time-line xdmf files

> @tkoyama010 here are the files [xdmf_files.zip](https://github.com/pyvista/pyvista/files/9037095/xdmf_files.zip)
> 
> I created 2 different files: one using checkpointing, the other without checkpointing.
> 
> Both are time series which contain 4 entries.
> 
> Here's the code to generate them:
> 
> ```python
> import fenics as f
> 
> mesh = f.UnitCubeMesh(8, 8, 8)
> V = f.FunctionSpace(mesh, "P", 1)
> u = f.Function(V)
> 
> expression = f.Expression("x[0]*x[1]*x[2] + t", degree=1, t=0)
> 
> export_file_wo_checkpoint = f.XDMFFile("no_checkpoint.xdmf")
> export_file_w_checkpoint = f.XDMFFile("checkpoint.xdmf")
> 
> for t in range(5):
>     expression.t = t
>     u.assign(f.interpolate(expression, V))
>     export_file_wo_checkpoint.write(u, t)
>     export_file_w_checkpoint.write_checkpoint(
>         u,
>         "label_u",
>         t,
>         f.XDMFFile.Encoding.HDF5,
>         append=(t != 0),
>     )
> ```

Reply:
Sorry, I got pulled into some other work, but have some bandwidth now to help with any work remaining. It looks like the tests have been implemented? Anything I can do to help?

Reply:
@gtpash Thank you. Please try using this branch of PyVista to see if it meets your requirements.

Reply:
Thanks for the update, I think I can get the class to meet my requirements, but only because I have a priori knowledge of the simulation times at which I saved output. The `set_active_time_value` method seems to work as intended. However, it would be nice to have the reader inherit from the `TimeReader` class. I do not know if there is an easy way to go about this as there don't seem to be built in methods for reading the time information (necessary to define `number_time_points`, `time_values`, etc) provided from VTK, at least as far as I can tell [according to the docs](https://vtk.org/doc/nightly/html/classvtkXdmfReader.html). However, one potential path would be to write a custom reader as was done for `.pvd` files with the `_PVDReader` class.